tests/phpunit/CRM/Contribute/Form/ContributionPageTranslationTest.php

```diff
@@ -0,0 +1,91 @@
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * Class CRM_Contribute_BAO_ContributionPageTranslationTest
+ * @group headless
+ */
+class CRM_Contribute_BAO_ContributionPageTranslationTest extends CiviUnitTestCase {
+
+  public function setUp() {
+    parent::setUp();
+    $this->_financialTypeID = 1;
+    $this->enableMultilingual();
+    CRM_Core_I18n_Schema::addLocale('fr_FR', 'en_US');
+  }
+
+  public function tearDown() {
+    global $dbLocale;
+    if ($dbLocale) {
+      CRM_Core_I18n_Schema::makeSinglelingual('en_US');
+    }
+  }
+
+  /**
+   * Create() method (create Contribution Page with Honor block)
+   */
+  public function testCreateHonor() {
+    CRM_Core_I18n::singleton()->setLocale('en_US');
+
+    $params = [
+      'title' => 'Test Contribution Page',
+      'financial_type_id' => $this->_financialTypeID,
+      'is_for_organization' => 0,
+      'for_organization' => ' I am contributing on behalf of an organization',
+      'goal_amount' => '400',
+      'is_active' => 1,
+      'honor_block_is_active' => 1,
+      'honor_block_title' => 'In Honor Title EN',
+      'honor_block_text' => 'In Honor Text EN',
+      'honoree_profile' => 13, // Honoree Individual
+      'soft_credit_types' => 1, // In Honor Of
+      'start_date' => '20091022105900',
+      'start_date_time' => '10:59AM',
+      'end_date' => '19700101000000',
+      'end_date_time' => '',
+      'is_credit_card_only' => '',
+    ];
+
+    $contributionpage = CRM_Contribute_BAO_ContributionPage::create($params);
+
+    // The BAO does not save these
+    $params['id'] = $contributionpage->id;
+    $params['honor_block_title'] = 'In Honor Title EN';
+    $params['honor_block_text'] = 'In Honor Text EN';
+
+    $form = $this->getFormObject('CRM_Contribute_Form_ContributionPage_Settings', $params, 'Settings');
+    $form->postProcess();
+
+    // Now update the page with In Honor (soft credit) text in French
+    CRM_Core_I18n::singleton()->setLocale('fr_FR');
+
+    $params['honor_block_title'] = 'In Honor Title FR';
+    $params['honor_block_text'] = 'In Honor Text FR';
+
+    $form = $this->getFormObject('CRM_Contribute_Form_ContributionPage_Settings', $params, 'Settings');
+    $form->postProcess();
+
+    $uf = $this->callAPISuccess('UFJoin', 'getsingle', [
+      'entity_id' => $contributionpage->id,
+      'module' => 'soft_credit',
+    ]);
+
+    $json = json_decode($uf['module_data'], TRUE);
+
+    $this->assertEquals($json['soft_credit']['en_US']['honor_block_title'], 'In Honor Title EN');
+    $this->assertEquals($json['soft_credit']['en_US']['honor_block_text'], 'In Honor Text EN');
+    $this->assertEquals($json['soft_credit']['fr_FR']['honor_block_title'], 'In Honor Title FR');
+    $this->assertEquals($json['soft_credit']['fr_FR']['honor_block_text'], 'In Honor Text FR');
```

Comment:
I like the test - just these params are backwards. Expected is the first param. When it fails it's briefly confusing.

Reply:
fixed